subprojects/gradle/build-properties/src/main/kotlin/com/avito/android/info/BuildPropertiesPlugin.kt

```diff
@@ -11,20 +9,33 @@ import org.gradle.kotlin.dsl.register
 
 open class BuildPropertiesPlugin : Plugin<Project> {
 
-    private lateinit var logger: CILogger
-
     override fun apply(project: Project) {
-        logger = project.ciLogger
+        registerPropertiesTask(project)
+        registerLegacyPropertiesTask(project)
+    }
+
+    private fun registerPropertiesTask(project: Project) {
+        val extension = project.extensions.create<BuildPropertiesExtension>("buildProperties")
+
+        val task = project.tasks.register<WriteProperties>("generateBuildProperties") {
```

Comment:
How will we depend on it? Should we declare contracts `val` with task names?

Reply:
`project.addPreBuildTasks(task)` - that's all we need